<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <!--Realsense CAM-->
  <!-- <include file="$(find realsense2_camera)/launch/rs_t265.launch" /> -->
  <!-- <include file="$(find hybrid_detection)/launch/rs_d435.launch" /> -->

  <!-- <arg name="markerId"        default="94"/> -->
  <arg name="marker1Id"         default="273"/>
  <!-- <arg name="marker2Id"         default="90"/> -->
  <arg name="markerSize"        default="0.137"/>    <!-- in m -->
<!--   <arg name="dct_normalization" default="True" />
  <arg name="dct_filter_size"   default="2" />
  <arg name="marker1_frame"     default="tag_1" />
  <arg name="marker2_frame"     default="tag_2" />
  <arg name="corner_refinement" default="SUBPIX"/> -->

  <arg name="markerSizeRCNN"    default="0.137"/>    <!-- in m -->
  <arg name="distancefocal"     default="720"/>  
  <arg name="maxnumberofboxes"  default="1"/>    
  <arg name="minimumconfidence" default="0.85"/>    <!-- in % -->
  <arg name="nameparent"        default="camera_odom_frame"/>

<!--   <node pkg="aruco_ros" type="double" name="aruco_double">
      <remap from="/camera_info" to="/camera/color/camera_info" />
      <remap from="/image" to="/camera/color/image_raw" />
      <param name="image_is_rectified" value="True"/>
      <param name="marker_size" value="$(arg markerSize)"/>
      <param name="marker_id1" value="$(arg marker1Id)"/>
      <param name="marker_id2" value="$(arg marker2Id)"/>
      <param name="normalizeImage" value="$(arg dct_normalization)"/>
      <param name="dct_components_to_remove" value="$(arg dct_filter_size)"/>
      <param name="parent_name" value="$(arg nameparent)"/>
      <param name="child_name1" value="$(arg marker1_frame)" />
      <param name="child_name2" value="$(arg marker2_frame)" />
      <param name="corner_refinement" value="$(arg corner_refinement)" />
  </node> -->

	<node pkg= "hybrid_detection" name="detect_rcnn_dataset" type="detect_rcnn_dataset.py"  output="screen"> 
      <remap from="/image" to="/camera/color/image_raw"/> <!--/cv_camera/image_rect_color or /bebop/image_raw -->
      <param name="parent_name" value="$(arg nameparent)"/>
      <param name="markerSize_RCNN" value="$(arg markerSizeRCNN)"/>
      <param name="distance_focal" value="$(arg distancefocal)"/>
      <param name="max_number_of_boxes" value="$(arg maxnumberofboxes)"/>
      <param name="minimum_confidence" value="$(arg minimumconfidence)"/>
  </node>

  <node pkg= "hybrid_detection" name="new_test_filter" type="new_test_filter.py"  output="screen"/>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find hybrid_detection)/rviz/rviz_config_fusion2.rviz"/>

</launch>
